[state_dict] Add the documentation to StateDictOptions

fegin opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #111261
* #111108
* #111107
* #111106

As title

Differential Revision: [D50289357](https://our.internmc.facebook.com/intern/diff/D50289357/)

fegin committed:
[state_dict] Add the documentation to StateDictOptions

As title

Differential Revision: [D50289357](https://our.internmc.facebook.com/intern/diff/D50289357/)

[ghstack-poisoned]

ZainRizvi reviewed (approved):
Approving pending success of the `build-docs-python` job
https://github.com/pytorch/pytorch/actions/runs/6499114499/job/17657996670

fegin committed:
Update on "[state_dict] Add the documentation to StateDictOptions"


As title

Differential Revision: [D50289357](https://our.internmc.facebook.com/intern/diff/D50289357/)

[ghstack-poisoned]

fegin committed:
Update on "[state_dict] Add the documentation to StateDictOptions"


As title

Differential Revision: [D50289357](https://our.internmc.facebook.com/intern/diff/D50289357/)

[ghstack-poisoned]
